web/src/engine/osk/src/keyboard-layout/oskLayerGroup.ts

```diff
@@ -75,30 +107,41 @@ export default class OSKLayerGroup {
       throw new Error(`Layer id ${layerId} could not be found`);
     }
 
-    // Kind of WET with VisualKeyboard.updateState, but oh well.
-    let priorLayer: OSKLayer = null;
-    if(layer.element.style.display == 'block') {
-      priorLayer = layer;
-    } else {
+    // Note:  we do NOT manipulate `._activeLayerId` here!  This is designed
+    // explicitly to be temporary.
+    if(layer.element.style.display != 'block') {
       for(let id in this.layers) {
         if(this.layers[id].element.style.display == 'block') {
-          priorLayer = this.layers[id];
+          const priorLayer = this.layers[id];
+          priorLayer.element.style.display = 'none';
         }
         this.layers[id].element.style.display = 'none';
       }
     }
     layer.element.style.display = 'block';
 
-    try {
-      return this.nearestKey(coord, layer);
-    } finally {
-      if(layer != priorLayer) {
+    /* As soon as control returns to the JS microtask queue, restore the original layer.
+     * We want to avoid doing it sooner in case another lookup occurs before the standard
+     * async reflow, as that could trigger expensive "layout thrashing" effects.
+     *
+     * In the case that a gesture-source's path needs to be remapped do a different layer,
```

Comment:
Should the "do" be a "to" ?

```suggestion
     * In the case that a gesture-source's path needs to be remapped to a different layer,
```